openedx/features/courseware/tests/test_pre_post_assessment_scores_api.py

```diff
@@ -0,0 +1,148 @@
+"""Unit tests for pre post assessment scores api"""
+
+from django.urls import reverse
+
+from rest_framework.test import APITestCase
+from rest_framework import status
+
+from openedx.features.courseware import constants
+from student.tests.factories import UserFactory, CourseEnrollmentFactory
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory, ItemFactory
+
+
+class PrePostAssessmentMixin(ModuleStoreTestCase):
+    """Mixin for pre/post assessments score tests"""
+
+    def create_course_and_user(self):
+        """
+        Create user and a simple course for testing pre/post assessments score api.
+        """
+        self.course = CourseFactory.create(display_name='test_course', number='100')
+        self.user = UserFactory.create(username='dummy', password='foo')
+        self.add_pre_post_assessment_grading_policy()
+
+    def create_chapter(self):
+        """Create chapter in course"""
+        return ItemFactory.create(
+            parent_location=self.course.location,
+            category='chapter',
+            display_name='Chapter'
+        )
+
+    def create_assessment_subsection(self, chapter, assessment_format=constants.PRE_ASSESSMENT_FORMAT):
+        """Create pre/post assessment in module/chapter"""
+        return ItemFactory.create(
+            parent_location=chapter.location,
+            display_name=assessment_format,
+            category='sequential',
+            metadata={'graded': True, 'format': assessment_format}
```

Comment:
We're using this method for both pre and post-assessment, so I guess we should not hard code graded to True, because pre-assessments are not graded.